### ISSUE TITLE
fix pytorch UserWarnings for softmax, item(), etc

### ISSUE THREAD

**Issue by christqoh** [facebookresearch/ParlAI#875]
Not an actual issue, just user warnings you might want to know about so it doesn't turn into one:

`ParlAI/projects/personachat/persona_seq2seq.py:1738: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  attn_weights = F.softmax(attn_w_premask)`

`ParlAI/projects/personachat/persona_seq2seq.py:1486: UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument.
  scores = F.log_softmax(scores)`

as found through running `python projects/convai2/baselines/profilememory/interactive.py`
